client.go

```diff
@@ -374,8 +369,18 @@ func getCIDEncoder() multibase.Encoder {
 func httperr(resp *http.Response, e error) error {
 	oerr, ok := e.(openapi.GenericOpenAPIError)
 	if !ok {
-		panic("wrong error type")
+		return e
 	}
+
+	ferr, ok := oerr.Model().(openapi.Failure)
+	if ok {
+		errJson, err := ferr.Error.MarshalJSON()
+		if err != nil {
+			return fmt.Errorf("failed to marshal Failure %w. Error was: {reason : %s, details : %s}", err, ferr.Error.GetReason(), ferr.Error.GetDetails())
+		}
+		return fmt.Errorf("%s", string(errJson))
```

Comment:
Should I also add in the status code (e.g. 401) or just have the reason + details?

Reply:
As this is the http client, it makes sense to include HTTP status code to help with debugging, but not a blocker.

Reply:
@lidel I removed the JSON marshalling here and did it manually. Added the HTTP status code text (i.e. "401 Unauthorized"), but could do the integer code to if you think that's nicer.

Reply:
I believe an integer will be easier to work with

Reply:
updated